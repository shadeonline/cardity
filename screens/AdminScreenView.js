
import React, { useEffect, useState } from 'react';
import { useNavigation } from '@react-navigation/core'
import { ScrollView, TouchableOpacity, StyleSheet, Text, View } from 'react-native';
import CreatePlanButton from '../components/CreatePlanButton.js';
import { auth } from '../firebase'
import { signOut } from 'firebase/auth';
import { AntDesign } from '@expo/vector-icons';
import { useRoute, useFocusEffect, useIsFocused } from '@react-navigation/native'
import { firebaseFetchProfile } from '../firebaseFunctions';

// Render cards according to the cards argument received
const AdminScreenView = () => {
  const isFocused = useIsFocused()
  const [profile, setProfile] = useState(null);
  const navigation = useNavigation()

  const handleSignOut = () => {
    signOut(auth)
      .then(() => {
        console.log("Signed out")
        navigation.replace("Login")
      })
      .catch(error => alert(error.message))
  }
  const fetchProfile = async () => {
    const profileData = await firebaseFetchProfile();
    setProfile(profileData);
  };

  useEffect(() => {
    if (isFocused) {
      fetchProfile();
    }
  }, [isFocused]);

  useFocusEffect(() => {
    navigation.setOptions({
      headerRight: () => (
        <TouchableOpacity
          onPress={() => {
            // Navigate to your camera screen when the icon is pressed
            navigation.navigate('Scanner'); // Replace with your actual camera screen
          }}
          style={{ marginRight: 20 }}
        >
          <AntDesign name="scan1" size={24} color="black" />
        </TouchableOpacity>
      ),
    });
  });

  return (
    <ScrollView>

      <View style={styles.container}>
        <CreatePlanButton text="Create Loyalty Plan" onPress={() => {
          navigation.navigate('Create Plan');
        }} />
      </View>

      {/* Sign Out */}
      <View style={styles.container}>
        <Text>Email: {auth.currentUser?.email}</Text>
        {profile && <Text>Name: {profile.name}</Text>}
        <TouchableOpacity
          onPress={handleSignOut}
          style={styles.button}
        >
          <Text style={styles.buttonText}>Sign out</Text>
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
};

export default AdminScreenView;



const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: '10%',
  },
  button: {
    backgroundColor: '#0782F9',
    width: '60%',
    padding: 15,
    borderRadius: 10,
    alignItems: 'center',
    marginTop: 40,
    justifyContent: 'center',
    alignItems: 'center'
  },
  buttonText: {
    color: 'white',
    fontWeight: '700',
    fontSize: 16,
  },
})