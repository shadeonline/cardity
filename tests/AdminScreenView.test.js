import React from 'react';
import { render } from 'react-native-testing-library';
import AdminScreenView from '../screens/AdminScreenView';
import { NavigationContainer } from '@react-navigation/native';


const mockNavigation = {
  setOptions: jest.fn(),
}; 

jest.mock('@react-navigation/stack', () => ({
  useFocusEffect: jest.fn(),
  useNavigation: () => mockNavigation, // Use the custom mock here
}));

jest.mock('../firebase', () => ({
  auth: {
    currentUser: {
      email: 'admin@example.com', // Mock the user's email here
    },
  },
}));

jest.mock('firebase/auth', () => ({
  signOut: jest.fn(), // Mock the signOut function here
}));

jest.mock('../firebaseFunctions', () => ({
  firebaseFetchProfile: jest.fn(),
  firebaseFetchLoyaltyProgramsByUser: jest.fn(),
}));

describe('AdminScreenView Component', () => {
  it('matches snapshot', () => {
    // Mock data
    const profile = {
      name: 'Admin Name',
    };
    const loyaltyPrograms = [
      {
        loyaltyCard: {
          cardName: 'Loyalty Card 1',
          color: 'blue',
          id: '1',
        },
        description: 'Description 1',
        rewards: {
          1: 'Reward 1',
          2: 'Reward 2',
        },
        storeName: 'Store 1',
        id: 'program1',
        task: 'Task 1',
      },
      // Add more loyalty programs as needed for testing
    ];

    // Render the AdminScreenView component with the mock data
    const { toJSON } = render(
      <NavigationContainer>
        <AdminScreenView
          profile={profile}
          loyaltyPrograms={loyaltyPrograms}
        />
      </NavigationContainer>
    );

    // Use Jest's expect to match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
