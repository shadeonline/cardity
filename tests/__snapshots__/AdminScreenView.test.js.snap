// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminScreenView Component matches snapshot 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#f0f0f0",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 8,
          }
        }
      >
        Welcome, 
        Admin
      </Text>
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 16,
          }
        }
      >
        admin@example.com
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e8e9ed",
          "borderRadius": 10,
          "height": 220,
          "justifyContent": "center",
          "marginBottom": 20,
          "opacity": 1,
          "width": 710,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 25,
          }
        }
      >
        +
      </Text>
      <Text
        style={
          {
            "fontSize": 25,
          }
        }
      >
        Create Loyalty Plan
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
          "marginTop": 16,
        }
      }
    >
      Loyalty Plans Created:
    </Text>
    <Text
      style={
        {
          "color": "#888",
          "fontSize": 16,
          "marginTop": 16,
          "textAlign": "center",
        }
      }
    >
      No loyalty programs created yet.
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ff3333",
          "borderRadius": 10,
          "marginTop": 16,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Sign out
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
